### DISCUSSION TITLE
Sidekiq with Ansible playbook failing on remote binaries

### DISCUSSION THREAD

**Discussion by kevinterrono** [sidekiq/sidekiq#6393] [Q&A]
We use Sidekiq to help manage automation jobs we develop using Ansible.  We have a web application that helps schedule and execute our jobs.  Part of our automation is to run a binary on a remote server, which runs without error if I just run this job
on the command line using native ansible tools or from the rails console.  However, the job only fails when Sidekiq runs the job.

        "stderr": "terminate called after throwing an instance of 'std::bad_alloc'\n  what():  std::bad_alloc",
        "stderr_lines": [
            "terminate called after throwing an instance of 'std::bad_alloc'",
            "  what():  std::bad_alloc"
        ],

A little more detail on what is happening, my playbook is calling a ksh script on the remote server and that ksh script is calling the binary inside of that.  At the time of the binary being called, is when I see the error.
		
The error seems to be related to some kind of memory limitation on the remote server.  Is this an issue that anyone has seen before or have an idea what I am missing here?  I'm trying to narrow down the possible causes and this is one of the only differences I can come up with on when this fails and succeeds. 

Version of rails: 7.0.8.3
Version of Sidekiq: 7.2.4

**Comment by mperham** [comment]
No idea, sorry.

**Reply by kevinterrono** [reply]
Ok.  Thank you.

**Reply by debow77** [reply]
@mperham one thing worth mentioning is that the same exact command works through the rails console. It's only when the same command is ran through the sidekiq job that it fails. It starts the script without issue, its only when the script executes the line invoking a command line tool `$cer_exe/scpview ${NODE} < /tmp/scp_commands.dat | tee -a ${logfile} > $1` that it then dies.

Was wondering if maybe it was a thread safe issue of some kind.

**Reply by mperham** [reply]
Those are two different environments. One has a TTY, the other does not. There can be ENV differences, etc.
